python/ray/node.py

```diff
@@ -397,14 +398,20 @@ def _prepare_socket_file(self, socket_path, default_prefix):
             socket_path (string): the socket file to prepare.
         """
         if socket_path is not None:
-            if os.path.exists(socket_path):
-                raise RuntimeError(
-                    "Socket file {} exists!".format(socket_path))
-            socket_dir = os.path.dirname(socket_path)
-            try_to_create_directory(socket_dir)
-            return socket_path
-        return self._make_inc_temp(
-            prefix=default_prefix, directory_name=self._sockets_dir)
+            if sys.platform != "win32":
+                if os.path.exists(socket_path):
+                    raise RuntimeError(
+                        "Socket file {} exists!".format(socket_path))
+                socket_dir = os.path.dirname(socket_path)
+                try_to_create_directory(socket_dir)
+            result = socket_path
+        elif sys.platform == "win32":
```

Comment:
Somehow the nesting structure of the ifs is not super clear at the moment. I think better would be:
```
if not windows:
  if socket_path is None:
    # create_or_runtime_error and return
  else:
    return self._make_ind_temp(...)
else:
  return socket_path or "tcp://{}:{}".format(self._localhost, self._get_unused_port())
```

What do you think?

Reply:
I agree that looks better, but it also changes the behavior when `socket_path` is e.g. an empty string. Is that fine?

Reply:
Although maybe it'll still be more readable if I mirror the top part with `if socket_path is None`, I could try that.